front-end/src/pages/CourseDetails.js

```diff
@@ -17,15 +17,17 @@ const CourseDetails = ({
 	price,
 	outline,
 }) => {
+	const [formTitle, setFormTitle] = useState(courseName);
```

Comment:
No need to have the courseName saved to another state variable.
Course Name should not change unless the user has navigated away to the page and then back using a link, then the courseName would be updated. You can just use courseName directly like you were doing before